src/midi/midi_fluidsynth.cpp

```diff
@@ -663,10 +663,57 @@ void MidiHandlerFluidsynth::ApplyChannelMessage(const std::vector<uint8_t>& msg)
 
 	// clang-format off
 	switch (status) {
-	case MidiStatus::NoteOff:         fluid_synth_noteoff(         synth.get(), channel, msg[1]);                 break;
-	case MidiStatus::NoteOn:          fluid_synth_noteon(          synth.get(), channel, msg[1], msg[2]);         break;
-	case MidiStatus::PolyKeyPressure: fluid_synth_key_pressure(    synth.get(), channel, msg[1], msg[2]);         break;
-	case MidiStatus::ControlChange:   fluid_synth_cc(              synth.get(), channel, msg[1], msg[2]);         break;
+	case MidiStatus::NoteOff:         fluid_synth_noteoff(     synth.get(), channel, msg[1]);         break;
+	case MidiStatus::NoteOn:          fluid_synth_noteon(      synth.get(), channel, msg[1], msg[2]); break;
+	case MidiStatus::PolyKeyPressure: fluid_synth_key_pressure(synth.get(), channel, msg[1], msg[2]); break;
+
+	case MidiStatus::ControlChange: {
+		const auto controller = msg[1];
+		const auto value = msg[2];
+
+		if (controller == MidiController::Portamento ||
+			controller == MidiController::PortamentoTime ||
+			controller == MidiController::PortamentoControl) {
+
```

Comment:
maybe avoid the empty block?

```cpp
if (controller != MidiController::Portamento &&
	controller != MidiController::PortamentoTime &&
	controller != MidiController::PortamentoControl) {
        // ...
	fluid_synth_cc(synth.get(), channel, controller, value);
}
```

Reply:
Yeah, I guess I like it better this way as it more resembles a case or when block.

If blah, that's a NOP
Else do stuff

I have a feeling we'll need to add further special cases in the future too.

Reply:
definitely disagree not enough to force the issue!